launch/launch/event_handlers/on_process_exit.py

```diff
@@ -54,7 +54,7 @@ def __init__(
         """Constructor."""
         from ..actions import ExecuteProcess  # noqa
         if not isinstance(target_action, (ExecuteProcess, type(None))):
-            raise RuntimeError("OnProcessExit requires an 'ExecuteProcess' action as the target")
+            raise TypeError("OnProcessExit requires an 'ExecuteProcess' action as the target")
```

Comment:
Some classes used `TypeError` for this, others used `RuntimeError`. I made them all consistent and used `TypeError`.